rest/route/middleware.go

```diff
@@ -494,3 +497,76 @@ func urlVarsToDistroScopes(r *http.Request) (string, int, error) {
 func superUserResource(_ *http.Request) (string, int, error) {
 	return evergreen.SuperUserPermissionsID, http.StatusOK, nil
 }
+
+type EventLogPermissionsMiddleware struct{}
+
+func (m *EventLogPermissionsMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	ctx := r.Context()
+	vars := gimlet.GetVars(r)
+	var resource string
+	var status int
+	var err error
+	resourceType := strings.ToUpper(vars["resource_type"])
+	opts := gimlet.PermissionOpts{}
+	switch resourceType {
+	case event.ResourceTypeTask:
+		resource, status, err = urlVarsToProjectScopes(r)
+		opts.ResourceType = evergreen.ProjectResourceType
+		opts.Permission = evergreen.PermissionTasks
+		opts.RequiredLevel = evergreen.TasksView.Value
+	case model.EventResourceTypeProject:
+		resource, status, err = urlVarsToProjectScopes(r)
+		opts.ResourceType = evergreen.ProjectResourceType
+		opts.Permission = evergreen.PermissionProjectSettings
+		opts.RequiredLevel = evergreen.ProjectSettingsView.Value
+	case event.ResourceTypeDistro:
+		fallthrough
+	case event.ResourceTypeScheduler:
+		resource, status, err = urlVarsToDistroScopes(r)
+		opts.ResourceType = evergreen.DistroResourceType
+		opts.Permission = evergreen.PermissionHosts
+		opts.RequiredLevel = evergreen.HostsView.Value
+	case event.ResourceTypeHost:
+		resource, status, err = urlVarsToDistroScopes(r)
+		opts.ResourceType = evergreen.DistroResourceType
+		opts.Permission = evergreen.PermissionDistroSettings
+		opts.RequiredLevel = evergreen.DistroSettingsView.Value
+	case event.ResourceTypeAdmin:
+		resource = evergreen.SuperUserPermissionsID
+		opts.ResourceType = evergreen.SuperUserResourceType
+		opts.Permission = evergreen.PermissionAdminSettings
+		opts.RequiredLevel = evergreen.AdminSettingsEdit.Value
+	default:
+		http.Error(rw, fmt.Sprintf("%s is not a valid resource type", resourceType), http.StatusBadRequest)
```

Comment:
should these errors be gimlet.ErrorMessage types? just for consistency?

Reply:
I don't see anything called ErrorMessage in gimlet :(